src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -47,10 +50,13 @@ public void printBasket() {
 
     public boolean checkingProductInTheCart(Product product) {
         boolean sign = false;
-        for (Product element : this.basket) {
-            if (element != null && element.getName() == product.getName()) {
-                sign = true;
-                return sign;
+        for (Map.Entry<String, LinkedList<Product>> entry : this.basket.entrySet()) {
```

Comment:
Тут можно упростить:
```suggestion
       return basket.containsKey(product));
```

Reply:
Спасибо! )